packages/indexer-agent/src/network.ts

```diff
@@ -389,18 +392,18 @@ export class Network {
         // Legacy transaction type
         // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
         if (feeData.gasPrice!.toNumber() >= this.baseFeePerGasMax) {
-          if (attempt == 1) {
+          if (attempt === 1) {
             logger.warning(
               `Max gas price has been reached, waiting until gas price estimates fall below to resume transaction execution.`,
               {
                 baseFeePerGasMax: this.baseFeePerGasMax,
-                currentGasPriceEstimate: feeData.gasPrice,
+                currentGasPriceEstimate: feeData.gasPrice.toNumber(),
```

Comment:
I believe you need a non-null assertion here, so it would be `feeData.gasPrice!.toNumber()`